.github/workflows/linux.yml

```diff
@@ -179,5 +179,19 @@ jobs:
     - name: Build & Install
       run: |
         ./configure --dim 2
-        make -j2
+        make -j2 WARN_ALL=TRUE
```

Comment:
do you also want to add this to the 1D case above? :)

Reply:
Another idea: we can add `-Werror` in CI to check PRs do not introduce warnings for selected compilers. Just a thought :)

Reply:
There are too many warnings in 1D we need to fix. :(  We do not fully support 1D.  So in many places we just have a function with empty body.  As for `-Werror`, we still have some warnings.  One we get rid of them, we can add that to a few CI tests.